openupgradelib/openupgrade.py

```diff
@@ -3261,17 +3261,43 @@ def chunked(records, single=True):
     """Memory and performance friendly method to iterate over a potentially
     large number of records. Yields either a whole chunk or a single record
     at the time. Don't nest calls to this method."""
-    if version_info[0] > 10:
-        invalidate = records.env.cache.invalidate
-    elif version_info[0] > 7:
-        invalidate = records.env.invalidate_all
+    # Warning on possibly untested versions where chunked might not work as expected
+    if version_info[0] > 17:  # unreleased version at this time
+        logger.warning(
+            f"chunked method might not work as expected in Odoo {version_info[0]}. "
+            "Please report any issue you may find and consider bumping this warning "
+            "to the next version otherwise."
+        )
```

Comment:
How does this relate to the rest of OpenUpgrade? Would we need a similar warning for every method?

Reply:
Quoting @pedrobaeza here: https://github.com/OCA/openupgradelib/issues/362#issuecomment-2023092055

> Well, this is the only method where it mixes ORM. The rest is thought for handling database things, so I don't think the rest are affected.

I don't think I have enough knowledge of this tool to answer this.

I was just bitten by the fact that I tried to use this method in `16.0` and got corrupted data as a result because of this silent error 😓 My motivation is preventing other people from going through the same in future versions, in case something like this changes again.